### PR TITLE
decouple from hyrax

### PR DESCRIPTION
**Pull request by labradford**

### CODE REVIEW

File: app/jobs/bulkrax/importer_job.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 module Bulkrax
   class ImporterJob < ApplicationJob
     queue_as :import
+    require 'csv'
```
Comment (orangewolf) [inline app/jobs/bulkrax/importer_job.rb:6]: require in this location will cause reloading issues. it also is almost certainly not nessisary

File: app/controllers/bulkrax/importers_controller.rb [hunk 2]
```diff
@@ -35,7 +35,10 @@ def show
       elsif defined?(::Hyrax)
         add_importer_breadcrumbs
         add_breadcrumb @importer.name
-
+        @work_entries = @importer.entries.where(type: @importer.parser.entry_class.to_s).page(params[:work_entries_page]).per(30)
```
Comment (orangewolf) [inline app/controllers/bulkrax/importers_controller.rb:38]: what am I missing? it really looks like the elseif and else are identical here. wouldn't it be better to just put an `end` on line 37?

File: app/models/bulkrax/csv_entry.rb [hunk 3]
```diff
@@ -67,9 +67,9 @@ def establish_factory_class
     end
 
     def add_metadata_for_model
-      if factory_class == Collection
+      if defined?(Hyrax) && factory_class == Collection
```
Comment (orangewolf) [inline app/models/bulkrax/csv_entry.rb:70]: what if the factory class is collection and you are not in hyrax?

File: app/models/concerns/bulkrax/export_behavior.rb [hunk 4]
```diff
@@ -27,8 +27,8 @@ def hyrax_record
     def filename(file_set)
       return if file_set.original_file.blank?
       fn = file_set.original_file.file_name.first
-      mime = Mime::Type.lookup(file_set.original_file.mime_type)
-      ext_mime = MIME::Types.of(file_set.original_file.file_name).first
+      mime = Marcel::MimeType.for(file_set.original_file.mime_type)
```
Comment (orangewolf) [inline app/models/concerns/bulkrax/export_behavior.rb:30]: 👍

File: lib/bulkrax.rb [hunk 5]
```diff
@@ -136,6 +137,10 @@ class << self
     self.qa_controlled_properties = %w[rights_statement license]
   end
 
+  def self.file_model_name
+    AcdaFile
```
Comment (orangewolf) [inline lib/bulkrax.rb:141]: this does not seem like the right default

File: app/parsers/bulkrax/csv_parser.rb [hunk 6]
```diff
@@ -192,12 +192,12 @@ def current_record_ids
 
       case importerexporter.export_from
       when 'all'
-        @work_ids = ActiveFedora::SolrService.query("has_model_ssim:(#{Hyrax.config.curation_concerns.join(' OR ')}) #{extra_filters}", method: :post, rows: 2_147_483_647).map(&:id)
+        @work_ids = ActiveFedora::SolrService.query("has_model_ssim:(#{Bulkrax.curation_concerns.join(' OR ')}) #{extra_filters}", method: :post, rows: 2_147_483_647).map(&:id)
         @collection_ids = ActiveFedora::SolrService.query("has_model_ssim:Collection #{extra_filters}", method: :post, rows: 2_147_483_647).map(&:id)
-        @file_set_ids = ActiveFedora::SolrService.query("has_model_ssim:FileSet #{extra_filters}", method: :post, rows: 2_147_483_647).map(&:id)
+        @file_set_ids = ActiveFedora::SolrService.query("has_model_ssim:#{Bulkrax.file_model_name.to_s} #{extra_filters}", method: :post, rows: 2_147_483_647).map(&:id)
```
Comment (alishaevn) [inline app/parsers/bulkrax/csv_parser.rb:197]:
```suggestion
        @file_set_ids = ActiveFedora::SolrService.query("has_model_ssim:#{Bulkrax.file_model_name} #{extra_filters}", method: :post, rows: 2_147_483_647).map(&:id)
```

Comment (alishaevn) [reply app/parsers/bulkrax/csv_parser.rb:197]: since `value.to_s` and `"#{value}"` are equivalent.

File: app/views/bulkrax/entries/show.html.erb [hunk 7]
```diff
@@ -33,7 +33,7 @@
 
     <p class='bulkrax-p-align'>
       <% if @importer.present? %>
-        <% factory_record = @entry.factory.find %>
+        <% factory_record = nil #@entry.factory.find %>
```
Comment (alishaevn) [inline app/views/bulkrax/entries/show.html.erb:36]: what's this change about?

File: app/services/bulkrax/remove_relationships_for_importer.rb [hunk 8]
```diff
@@ -68,6 +69,7 @@ def break_relationships!
       end
     end
 
+    # @todo(bjustice) - remove hyrax reference
```
Comment (alishaevn) [inline app/services/bulkrax/remove_relationships_for_importer.rb:72]: I've seen a few of these comments. is there more to do or can they be deleted?

File: app/models/bulkrax/csv_entry.rb [hunk 9]
```diff
@@ -67,9 +67,9 @@ def establish_factory_class
     end
 
     def add_metadata_for_model
-      if factory_class == Collection
-        add_collection_type_gid
-      elsif factory_class == FileSet
+      if defined?(::Collection) && factory_class == ::Collection
+        add_collection_type_gid if defined?(Hyrax)
```
[the remark this answers is not in the document]
Comment (sephirothkod) [reply app/models/bulkrax/csv_entry.rb:71]: sure, but we don't know what that is for each different system. maybe I could move the check into the add_collection_type_gid method instead?

File: app/parsers/bulkrax/csv_parser.rb (hunk 6 above)
Comment (sephirothkod) [reply app/parsers/bulkrax/csv_parser.rb:197]: ooh yup!

File: app/models/bulkrax/csv_entry.rb (hunk 9 above)
[the remark this answers is not in the document]
Comment (alishaevn) [reply app/models/bulkrax/csv_entry.rb:71]: I was thinking more like "if hyrax do this, if not do that". even if the "that" references an empty method. it at least lets the implementer know that they need to figure out what to do in their non hyrax use case.

File: app/views/bulkrax/entries/show.html.erb (hunk 7 above)
Comment (sephirothkod) [reply app/views/bulkrax/entries/show.html.erb]: it was a temporary workaround that i forgot to remove. should be good now.

File: app/models/bulkrax/csv_entry.rb (hunk 9 above)
[the remark this answers is not in the document]
Comment (sephirothkod) [reply app/models/bulkrax/csv_entry.rb:71]: sure that works too. I was thinking if I moved the hyrax check into the add_collection_type_gid method then a non hyrax user could just override that method.

File: app/services/bulkrax/remove_relationships_for_importer.rb (hunk 8 above)
Comment (sephirothkod) [reply app/services/bulkrax/remove_relationships_for_importer.rb:72]: There is more to do that is not vital to this all working for importers. the idea is that once we test exporting and relationships we can come back and make the needed changes.

Comment (sephirothkod) [reply app/services/bulkrax/remove_relationships_for_importer.rb:72]: agreeing to just have a comment here to clarify that if users want to add additional (or non hyrax) metadata here is the place to do it for collections.

File: app/parsers/bulkrax/parser_export_record_set.rb [hunk 10]
```diff
@@ -87,14 +87,14 @@ def each
       #
       # @see #file_sets
       def candidate_file_set_ids
-        @candidate_file_set_ids ||= works.flat_map { |work| work.fetch("file_set_ids_ssim", []) }
+        @candidate_file_set_ids ||= works.flat_map { |work| work.fetch("#{Bulkrax.file_model_name.to_s.underscore}_ids_ssim", []) }
       end
 
-      # @note Specifically not memoizing this so we can merge values without changing the object.
+      # @note Specifically not memorizing this so we can merge values without changing the object.
```
Comment (alishaevn) [inline app/parsers/bulkrax/parser_export_record_set.rb:93]:
```suggestion
      # @note Specifically not memoizing this so we can merge values without changing the object.
```

File: app/views/bulkrax/exporters/_form.html.erb [hunk 11]
```diff
@@ -52,16 +52,13 @@
...
+  <%= form.input :export_source_worktype,
     label: t('bulkrax.exporter.labels.worktype'),
     required: true,
     prompt: 'Select from the list',
     label_html: { class: 'worktype export-source-option hidden' },
     input_html: { class: 'worktype export-source-option hidden form-control' },
-    collection: Hyrax.config.curation_concerns.map {|cc| [cc.to_s, cc.to_s] } %>
-  <% end %>
-  
+    collection: Bulkrax.curation_concerns.map {|cc| [cc.to_s, cc.to_s] } %>
```
Comment (alishaevn) [inline app/views/bulkrax/exporters/_form.html.erb:61]:
```suggestion
    collection: Bulkrax.curation_concerns.map { |cc| [cc.to_s, cc.to_s] } %>
```

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: app/parsers/bulkrax/parser_export_record_set.rb [hunk 12]
```diff
@@ -87,14 +87,14 @@ def each
...
       #
       # No sense attempting to query for more than the limit.
       def query_kwargs
-        { fl: "id,file_set_ids_ssim", method: :post, rows: row_limit }
+        { fl: "id,#{Bulkrax.file_model_name.to_s.underscore}_ids_ssim", method: :post, rows: row_limit }
```
[the remark this answers is not in the document]
Comment (sephirothkod) [reply app/parsers/bulkrax/parser_export_record_set.rb:97]: This doesn't work. if file model name is set to the model it errors when trying to convert to underscore case. I originally did it without the to_s and it failed, that's why I added it here.

File: app/parsers/bulkrax/parser_export_record_set.rb (hunk 10 above)
Comment (sephirothkod) [reply app/parsers/bulkrax/parser_export_record_set.rb:93]: I bet that this was my spellchecker. Never works when it is supposed to, changes things when it shouldn't

File: app/parsers/bulkrax/parser_export_record_set.rb [hunk 13]
```diff
@@ -87,14 +87,14 @@ def each
       #
       # @see #file_sets
       def candidate_file_set_ids
-        @candidate_file_set_ids ||= works.flat_map { |work| work.fetch("file_set_ids_ssim", []) }
+        @candidate_file_set_ids ||= works.flat_map { |work| work.fetch("#{Bulkrax.file_model_name.to_s.underscore}_ids_ssim", []) }
```
[the remark this answers is not in the document]
Comment (sephirothkod) [reply app/parsers/bulkrax/parser_export_record_set.rb:90]: see comment below

File: app/jobs/bulkrax/schedule_relationships_job.rb [hunk 14]
```diff
@@ -9,7 +9,7 @@ def perform(importer_id:)
       return reschedule(importer_id) unless pending_num.zero?
 
       importer.last_run.parents.each do |parent_id|
-        CreateRelationshipsJob.perform_later(parent_identifier: parent_id, importer_run_id: importer.last_run.id)
+        Bulkrax.relationship_job.perform_later(parent_identifier: parent_id, importer_run_id: importer.last_run.id)
```
Comment (bkiahstroud) [inline app/jobs/bulkrax/schedule_relationships_job.rb:12]: Not a big deal, but I like the idea of this being called `relationship_job_class`

File: app/matchers/bulkrax/application_matcher.rb [hunk 15]
```diff
@@ -101,6 +101,7 @@ def extract_model(src)
     end
 
     # Only add valid resource types
+    # @todo(bjustice) - remove hyrax reference NOT USED?
```
Comment (bkiahstroud) [inline app/matchers/bulkrax/application_matcher.rb:104]: Is this still a TODO?

File: app/parsers/bulkrax/xml_parser.rb [hunk 16]
```diff
@@ -78,16 +79,19 @@ def file_paths
...
           file_paths.select do |f|
-            MIME::Types.type_for(f).include?('application/xml') &&
-              f.include?("import_#{importerexporter.id}")
+            good_file_type?(f) && f.include?("import_#{importerexporter.id}")
           end
         end
     end
 
+    def good_file_type?(path)
+      File.extname(path) || ::Marcel::MimeType.for(path).include?('application/xml')
```
Comment (bkiahstroud) [inline app/parsers/bulkrax/xml_parser.rb:92]: What is `File.extname(path)` doing here?

File: app/services/bulkrax/remove_relationships_for_importer.rb [hunk 17]
```diff
@@ -50,12 +50,13 @@ def initialize(entries:, progress_bar:)
 
     attr_reader :entries, :progress_bar
 
+    # @todo(bjustice) - remove hyrax reference
```
Comment (bkiahstroud) [inline app/services/bulkrax/remove_relationships_for_importer.rb:53]: Is this still a TODO?

File: app/services/bulkrax/remove_relationships_for_importer.rb (hunk 8 above)
Comment (bkiahstroud) [inline app/services/bulkrax/remove_relationships_for_importer.rb:72]: Is this still a TODO?

File: app/matchers/bulkrax/application_matcher.rb (hunk 15 above)
Comment (sephirothkod) [reply app/matchers/bulkrax/application_matcher.rb:104]: I don't think this method is used at all in Bulkrax.

File: app/parsers/bulkrax/xml_parser.rb (hunk 16 above)
Comment (sephirothkod) [reply app/parsers/bulkrax/xml_parser.rb:92]: It's supposed to check the extension, except I didn't add it... fixed this in my latest commit.

File: app/services/bulkrax/remove_relationships_for_importer.rb (hunk 17 above)
Comment (sephirothkod) [reply app/services/bulkrax/remove_relationships_for_importer.rb:53]: ended up taking care of this.

File: app/services/bulkrax/remove_relationships_for_importer.rb (hunk 8 above)
Comment (sephirothkod) [reply app/services/bulkrax/remove_relationships_for_importer.rb:72]: ended up taking care of this.

File: app/jobs/bulkrax/schedule_relationships_job.rb (hunk 14 above)
Comment (sephirothkod) [reply app/jobs/bulkrax/schedule_relationships_job.rb]: I like that too, I changed the file model variable to use the same name format so it matches.

File: app/matchers/bulkrax/application_matcher.rb (hunk 15 above)
Comment (bkiahstroud) [reply app/matchers/bulkrax/application_matcher.rb:104]: Hm, I'm not sure. I've asked [in Slack](https://assaydepot.slack.com/archives/G0313NK5NMA/p1679593992121479)

**Pull request merged**